qiskit_experiments/characterization/qubit_spectroscopy.py

```diff
@@ -45,203 +48,62 @@ class SpectroscopyAnalysis(BaseAnalysis):
+            ),
+            plot_color="blue",
         )
+    ]
 
-    # pylint: disable=arguments-differ, unused-argument
-    def _run_analysis(
-        self,
-        experiment_data: ExperimentData,
-        data_processor: Optional[callable] = None,
-        amp_guess: Optional[float] = None,
-        sigma_guesses: Optional[List[float]] = None,
-        freq_guess: Optional[float] = None,
-        offset_guess: Optional[float] = None,
-        amp_bounds: Tuple[float, float] = (-1, 1),
-        sigma_bounds: Optional[Tuple[float, float]] = None,
-        freq_bounds: Optional[Tuple[float, float]] = None,
-        offset_bounds: Tuple[float, float] = (-2, 2),
-        plot: bool = True,
-        ax: Optional["AxesSubplot"] = None,
-        **kwargs,
-    ) -> Tuple[AnalysisResult, None]:
-        """Analyze the given data by fitting it to a Gaussian.
-
-        Args:
-            experiment_data: The experiment data to analyze.
-            data_processor: The data processor with which to process the data. If no data
-                processor is given a singular value decomposition of the IQ data will be
-                used for Kerneled data and a conversion from counts to probabilities will
-                be done if Discriminated data was measured.
-            amp_guess: The amplitude of the Gaussian function, i.e. :math:`a`. If not
-                provided, this will default to -1 or 1 depending on the measured values.
-            sigma_guesses: The guesses for the standard deviation of the Gaussian distribution.
-                If it is not given this will default to an array of ten
-                points linearly spaced between zero and width of the x-data.
-            freq_guess: A guess for the frequency of the peak :math:`x0`. If not provided
-                this guess will default to the location of the highest or lowest point of
-                the y-data depending on the y-data.
-            offset_guess: A guess for the magnitude :math:`b` offset of the fit function.
-                If not provided, the initial guess defaults to the median of the y-data.
-            amp_bounds: Bounds on the amplitude of the Gaussian function as a tuple of
-                two floats. The default bounds are (-1, 1).
-            sigma_bounds: Bounds on the standard deviation of the Gaussian function as a tuple
-                of two floats. The default values are (0, frequency range).
-            freq_bounds: Bounds on the center frequency as a tuple of two floats. The default
-                values are (min(frequencies) - df, max(frequencies) - df).
-            offset_bounds: Bounds on the offset of the Gaussian function as a tuple of two floats.
-                The default values are (-2, 2).
-            plot: If True generate a plot of fitted data.
-            ax: Optional, matplotlib axis to add the plot to.
-            kwargs: Trailing unused function parameters.
```

Comment:
Most of them are written in the superclass.
https://github.com/Qiskit/qiskit-experiments/blob/0729d6cb6e4349f77f9b2ba7b5af75df72dbd6be/qiskit_experiments/analysis/curve_analysis.py#L231-L273

Note that parameter-wise p0 and bound specification is changed to dictionary.

Reply:
We should at least document somewhere in this analysis class what happens to these parameters when the user does not specify a value. Otherwise the user will have to dig through the code to figure out what initial guesses and bounds are being used when he does not specify any. Can you add this information in the dosctrings of SpectroscopyAnalysis?

Reply:
I'm looking for an option that dynamically updates the docstring of experiment class (`.analysis_options`) based on documentation attached to each argument. Of course we can write these documents in every class docstring, but this means we need to manage docstring of every subclass when we add a new option to the superclass. I think this is very annoying and we cannot test this with linter. So I can imagine at some point people forget to sync docstring with superclass.

Reply:
Resolved by b416c80